### PR TITLE
Input checks

### PR DESCRIPTION
**Pull request by cgntnr**
* Input checks are completed for profile preferences pages.
* Hints are added for usability for text inputs.
* For Profile7, since we haven't done the linking part; button sends back to MainActivity in case of success. I've added TODOs and comments so that we correct these once linking activities are done.

### CODE REVIEW

**Review by Lollipoke — APPROVED** [review]
Really nothing to say about your code 👍

**Comment by cgntnr** [comment]
Unfortunately I had to comment out two tests for Passions page however Cirrus couldn't seem to perform clicks on chips on that page. The tests worked fine locally.

File: app/src/main/res/layout/set_profile_4_2.xml [hunk 1]
```diff
@@ -25,6 +25,7 @@
         android:layout_height="50dp"
         android:contentDescription="@string/please_specify_gender"
         android:ems="10"
+        android:hint="@string/more_hint"
```
Comment (r-ismer) [inline app/src/main/res/layout/set_profile_4_2.xml:28]: does the hint get covered by the contentDescription ? if it is the case remove one of the two

**Review by r-ismer — APPROVED** [review]
very good work ! looks very good to me. At some point we will have to check if the strings are actually used so that cirrus don't display warnings

File: app/src/main/res/layout/set_profile_4_2.xml (hunk 1 above)
Comment (cgntnr) [reply app/src/main/res/layout/set_profile_4_2.xml:28]: hint is the one that appears greyed out before you start to type something. contentDescription was for Accessibility if I'm not mistaken.

**Review by r-ismer — COMMENTED** [review]
everything is clear, looks good to me

**Pull request merged**